proxy/http/HttpSM.cc

```diff
@@ -6560,6 +6559,12 @@ HttpSM::kill_this()
   reentrancy_count--;
   ink_release_assert(reentrancy_count == 0);
 
+  // Delay the close of the user agent session, so the close session
+  // occurs after the close transaction
+  if (ua_session != NULL)
```

Comment:
What if the SM is killed before a tunnel is started (e.g. error responses before remap, origin connection failure etc)?